addons/account/models/account_payment_term.py

```diff
@@ -51,6 +51,11 @@ def _compute_fiscal_country_codes(self):
             allowed_companies = record.company_id or self.env.companies
             record.fiscal_country_codes = ",".join(allowed_companies.mapped('account_fiscal_country_id.code'))
 
+    @api.depends_context('company')
+    def _compute_currency_id(self):
```

Comment:
the `depends` is more important than the `depends_context` (you might change the company in the form view, but you can't change the context without reloading the form completely)
```suggestion
    @api.depends_context('company')
    @api.depends('company_id')
    def _compute_currency_id(self):
```